modules/Kconfig.tfm

```diff
@@ -0,0 +1,25 @@
+# TF-M/PSA Related
+
+# Copyright (c) 2019, 2020 Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+config BUILD_WITH_TFM
+	bool "Build with TF-M for Secure Execution"
```

Comment:
This option requires a help text. 
We need to explain what it means "to build with TF-M for Secure Execution":

For instance: shall this mean "Build a NON-SECURE Zephyr execution environment with TF-M for Secure Execution"?
If so I would add a "depends on TRUSTED_EXECUTION_NONSECURE"
and then this would mean that:
in order to build a non-secure Zephyr firmware with TF-M I need to
- select an "_ns" Zephyr board target (which selects TRUSTED_EXECUTION_NONSECURE"
- enable CONFIG_BUILD_WITH_TFM"
In order to build a

Reply:
Is this sufficiently clear, in your opinion?

```
config BUILD_WITH_TFM
	bool "Build with TF-M for Secure Execution"
	depends on TRUSTED_EXECUTION_NONSECURE
	help
	  This will cause the Zephyr application to be placed in the
	  non-secure processing environment, and a TF-M image will be
	  generated for the secure processing environment. Both
	  images, as well as the veneer object file that links them, will
	  be generated during the normal Zephyr build process.

	  When building with CONFIG_BUILD_WITH_TFM enabled, you must use
	  the "*_nonsecure" board variant (ex. "mps2_an521_nonsecure").
```

Reply:
Yeap, the main points are covered, let me work on an iteration (will post it here)

Reply:
```
config BUILD_WITH_TFM
	bool "Build with TF-M as the Secure Execution Environment"
	depends on TRUSTED_EXECUTION_NONSECURE
	help
	  When enabled, this option instructs the Zephyr build process to
          additionaly generate a TF-M image for the Secure Execution
          environment, along with the Zephyr image. The Zephyr image
          itself is to be executed in the Non-Secure Processing Environment.
          The required dependency on TRUSTED_EXECUTION_NONSECURE
          ensures that the Zephyr image is built as a Non-Secure image. Both
	  TF-M and Zephyr images, as well as the veneer object file that links
          them, are generated during the normal Zephyr build process.

          Note:
	    Building with the "*_nonsecure" BOARD variant (e.g.
            "mps2_an521_nonsecure") ensures that
            CONFIG_TRUSTED_EXECUTION_NONSECURE ie enabled.
```

Reply:
Maybe this one @microbuilder @tejlmand ? LAtest version is more explanatory around the dependency on TRUSTED_EXECUTION_NONSECURE.

Reply:
Updated with this, thanks for the suggestion. Agree it's clearer.